### DISCUSSION TITLE
Empty div unexpectedly inserted into body

### DISCUSSION THREAD

**Discussion by MagnusOxlund** [tinymce/tinymce#9533] [General]
An empty `div` element is being inserted into `body` after TinyMCE has been initialized. It has the classes `tox`, `tox-silver-sink`, and `tox-tinymce-aux`. I can't find any mention of it except for #5471 which doesn't reveal much.

Could someone explain what this `div` is and what it's doing in the markup?

<img width="506" alt="Screenshot 2024-04-05 at 18 02 41" src="https://github.com/tinymce/tinymce/assets/10921083/e50f3428-943a-4ffb-9b10-c3cfb825507c">

**Comment by MagnusOxlund** [comment]
Turns out it's "the sink", a part of TinyMCE that's responsible for drawing UI overlays when the user interacts with UI controls like buttons in the toolbar. It's inserted into the `body` by default.

If you would like to move it after TinyMCE has finished loading, you can use the `PostRender` hook:

```
/* Move "sink" out of body and into editor's parent node. */
editor.on('PostRender', function() {
    let container = editor.getContainer();
    let sink = document.querySelector('div.tox.tox-tinymce-aux');
    container.parentNode.appendChild(sink);

    /* Match sink width to parent to prevent blowout. */
    sink = "width: 100%; position: relative;"
});
```

Another option is to set the [`ui_mode`](https://www.tiny.cloud/docs/tinymce/6/ui-mode-configuration-options/) init parameter to `split`, which moves the sink into the editor's parent, just like the code sample above. It's mentioned in #7868.

**Reply by TheSpyder** [reply]
We don't recommend moving the sink manually. Many parts of the code use the assumed location to position UI elements placed inside it.

It exists as a separate concept to the editor so we can guarantee modal dialogs and long menus aren't constrained by potential CSS limits around the editor (such as `overflow`). We added `ui_mode` after it became clear that the end of the body doesn't work for all scenarios.

**Reply by MagnusOxlund** [reply]
Thank you for the warning. Setting `ui_mode: 'split'` does the job in my case, so I'll stick with that. 

Is there a reason the sink is not mentioned in the docs?

Maybe I've missed it, but is doesn't look like there's anything that ties `div.tox.tox-tinymce-aux` to the [`ui_mode` chapter](https://www.tiny.cloud/docs/tinymce/7/ui-mode-configuration-options/).

**Reply by TheSpyder** [reply]
No, it's part of the way our UI was built and shouldn't be something developers need to think about.

**Reply by MagnusOxlund** [reply]
Like I mentioned in #7868, the sink will interfere with a grid layout on `body`, triggering a row. This happens with the default configuration, so I think developers would appreciate being able to tie the mysterious `div` to the setting that fixes that issue.

Here's an example:
```lang-html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <script src="https://cdn.tiny.cloud/1/no-api-key/tinymce/7/tinymce.min.js" referrerpolicy="origin"></script>
    <title>Document</title>
    <script>
        tinymce.init({
            selector: '#tinymce',
            ui_mode: 'combined' // this is the default value
        });
    </script>
    <style>
        body {
            margin: 0;
            height: 100vh;
            display: grid;
            grid-template-rows: 1fr 1fr 1fr;
            grid-auto-rows: 1fr;
        }

        header, main, footer {
            background-color: teal;
        }
    </style>
</head>
<body>
    <header>header</header>
    <main>
        <form action="#">
            <textarea id="tinymce"></textarea>
        </form>
    </main>
    <footer>footer</footer>
</body>
</html>
```
